test/server-side-test.js

```diff
@@ -1,81 +1,233 @@
+        .then((user) => user.attributes.password_hash)
+          .then((passwordHash) => {
+            bcrypt.compare('hatch1', passwordHash, (err, match) => {
+              expect(match).to.be.true;
+              done();
+            })
+          })
+    })
   })
 
-  describe('Data insertion', function() {
+  describe('Data retrieval', function() {
+    it('should retrieve user by email (test 1)', (done) => {
+      User.getUserByEmail({ email: 'John@gmail.com'}, usersModel)
+        .then((user) => {
+          should.exist(user);
+          expect(user.attributes.first_name).to.equal('John');
+          expect(user.attributes.last_name).to.equal('Doe');
+          expect(user.attributes.is_active).to.equal(true);
+          done();
+        }, done);
+    });
 
-    it('should return correct first name given an email', function() {
-      return new Users({email: 'nguyenaiden@gmail.com'}).fetch().then(function(model) {
-        return expect(model.attributes.first_name).to.equal('Duy');
-      });
+    it('should retrieve user by email (test 2)', (done) => {
+      User.getUserByEmail({ email: 'Jane@gmail.com'}, usersModel)
+        .then((user) => {
+          should.exist(user);
+          expect(user.attributes.first_name).to.equal('Jane');
+          expect(user.attributes.last_name).to.equal('Doe');
+          expect(user.attributes.is_active).to.equal(true);
+          done();
+        }, done);
+    });
+    
+    it ('should retrieve user by ID (test 1)', (done) => {
+      User.getUserById({ id: 1 }, usersModel)
+        .then((user) => {
+          should.exist(user);
+          expect(user.attributes.first_name).to.equal('John');
+          expect(user.attributes.last_name).to.equal('Doe');
+          expect(user.attributes.is_active).to.equal(true);
+          done();
+        }, done)
+    });
+
+    it ('should retrieve user by ID (test 2)', (done) => {
+      User.getUserById({ id: 2 }, usersModel)
+        .then((user) => {
+          should.exist(user);
+          expect(user.attributes.first_name).to.equal('Jane');
+          expect(user.attributes.last_name).to.equal('Doe');
+          expect(user.attributes.is_active).to.equal(true);
+          done();
+        }, done)
+    });
+  })
+
+
+  describe('Data update', function() {
+    beforeEach(() => {
+      this.userUpdateParams2 = {
+        id: 2,
+        first_name: 'Jane',
+        last_name: 'Doe',
+        password: 'smallowl',
+        phone_number: '+14441114444',
+        email: 'Jane@yahoo.com'
+      };
+
+      this.userUpdateParams1 = {
+        id: 1,
+        first_name: 'John',
+        last_name: 'Wilson',
+        password: 'bigowl',
+        phone_number: '+14441114444',
+        email: 'John@yahoo.com'
+      };
+    });
+
+    it('should update user email by ID', (done) => {
+      User.updateUserById(this.userUpdateParams2, usersModel)
+        .then((user) => user.attributes.email)
```

Comment:
Be consistent with the arrow function parameter. Sometimes you wrap them in `()`, sometimes you don't (`(user)` vs `email` on the next line)